File: 0xxx/07xx/070x/0704.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
const search = function (nums, target) {
  if (nums.length === 0) return -1
  let min = -1
  let max = nums.length

  while (max - min > 1) {
    if (nums[Math.floor((min + max) / 2)] < target) {
      min = Math.floor((min + max) / 2)
    } else if (nums[Math.floor((min + max) / 2)] > target) {
      max = Math.floor((min + max) / 2)
    } else {
      return Math.floor((min + max) / 2)
    }
  }
  return -1
}

File: 0xxx/07xx/070x/0704.py
class Solution:
  def search(self, nums: List[int], target: int) -> int:
    if len(nums) == 0: return -1
    
    min = -1
    max = len(nums)
    while max - min > 1:
      mid = (max + min) // 2
      if nums[mid] > target:
        max = mid
      elif nums[mid] < target:
        min = mid
      else:
        return mid
    return -1